ast/compile_test.go

```diff
@@ -1959,6 +1959,24 @@ func TestCompilerCheckRuleConflictsDotsInRuleHeads(t *testing.T) {
 				p.q.s = "x" { true }
 				`),
 		},
+		{
+			note: "multi-value rule with other rule overlap",
+			modules: modules(
+				`package pkg
+				p[v] { v := ["a", "b"][_] }
+				p.q := 42
```

Comment:
Can we add a test like this? Or do we already have one?

```
p[x] { ... }
p.foo { ... }
```

That's used in some product bundles, iirc.